doc/adrs/193_add_manage_visibility_actions_api.adoc

```diff
@@ -0,0 +1,81 @@
+= ADR-193 - Add manage visibility actions API
+
+== Context
+
+A node action is used to open a modal in order to manage the visibility of the children of a node.
+
+== Decision
+
+We will add an API for retreiving and executing action from the backend.
+
+=== Frontend
+
+We will add 2 hooks that will be used to execute the needed query and mutation.
+
+[code,typescript]
+----
+const { manageVisibilityActions } = useManageVisibilityActions(diagramElementId);
+const { invokeManageVisibilityAction } = useInvokeManageVisibilityAction(diagramElementId);
+----
+
+=== GraphQL shema
+
+We will reuse the existing Action type.
+
+----
+type Action {
+  id: ID!
+  tooltip: String!
+  iconURLs: [String!]!
+}
+----
```

Comment:
Don't reuse anything, you would create some coupling between two distinct features:

- A list item used specifically to update the visibility of some diagram elements
- A button visible on over on a node

Both feature may not evolve in the same direction, do not create this kind of coupling

Reply:
You don't really need a tooltip or an iconURLs for example but you are required to have a label. Never create useless coupling.